osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -95,88 +95,39 @@ public override double Calculate(Dictionary<string, double> categoryRatings = nu
-                }
-                return JumpDistanceArray;
-            }
-        }
-        catch (IOException e)
+
+        public double[] Main()
         {
-            double[] errorval = {1,2,4};
-            Console.WriteLine("The file could not be read:");
-            Console.WriteLine(e.Message);
-            return errorval;
-        }
-    }
-public double Main2()
-    {
-        try
-        {   // Open the text file using a stream reader.
-            string tempstor = Path.GetTempPath();
-            string jda2path = tempstor + "jda2.txt";
-            using (StreamReader sr = new StreamReader(jda2path))
-            {
-            // Read the stream to a string, and write the string to the console.
-                double sumstrain = 0;
-                string a = sr.ReadToEnd();
-                a = a.TrimEnd(',');
-                string[] StrainTimeArray2 = a.Split(',');
-                for (int i = 0; i < StrainTimeArray2.Length; i++)
-                {
-                    double intermres2 = double.Parse(StrainTimeArray2[i],CultureInfo.InvariantCulture);
-                    sumstrain = sumstrain + intermres2;
-                }
-                return sumstrain;
-            }
+            return Attributes.JumpDistances;
         }
-        catch (IOException e)
+
+        public double Main2()
         {
-            double errorval = 2;
-            Console.WriteLine("The file could not be read:");
-            Console.WriteLine(e.Message);
-            return errorval;
+            return Attributes.StrainTimes.Sum();
         }
-    }
+
         private double computeAimValue()
         {
             double[] JumpDistanceArray = Main();
             double finalsumstrain = Main2();
             // var Aim = new Aim();
             double average = JumpDistanceArray.Average();
             double sumOfSquaresOfDifferences = JumpDistanceArray.Select(val => (val - average) * (val - average)).Sum();
-            double finalsd = Math.Sqrt(sumOfSquaresOfDifferences / JumpDistanceArray.Length); 
+            double finalsd = Math.Sqrt(sumOfSquaresOfDifferences / JumpDistanceArray.Length);
             double bpm = (60000 * countHitCircles) / finalsumstrain;
-            double finalsd2 = (bpm/375) * finalsd;
+            double finalsd2 = (bpm / 375) * finalsd;
             double finalsdpp = 0;
+
             if (countMiss == 0)
             {
-                  finalsdpp = finalsdpp+(finalsd2/stdevconst);
-                  if (finalsdpp >= 1)
-                  {
-                  finalsdpp = (-1)*(Log (finalsdpp, 0.000001));
-                  }
+                finalsdpp = finalsdpp + (finalsd2 / stdevconst);
+
+                if (finalsdpp >= 1)
+                {
+                    finalsdpp = (-1) * (Log(finalsdpp, 0.000001));
+                }
             }
+
             double rawAim = Attributes.AimStrain + finalsdpp;
             // Always clear temp file after end sd workflow.
```

Comment:
Do we still need these lines? (Temp .db storage removal)

Reply:
Ah I missed those, no they're not needed.